### PR TITLE
Castle Siege Away Mission (DNM)

### PR DESCRIPTION
**Pull request by MMMiracles**
I remember someone (kor maybe?) suggesting an away mission that was essentially ghosts in an eternal medieval war or something. I made it because that sounds fun. PRing so I don't forget about again.

Two kingdoms, the Totharenes and Yathereans (these are good names fuck you). They apparently pissed off a wizard in the medieval times or something so now they're stuck in an eternal war against each other on an ice rock.

Currently has a battlefield that players will spawn somewhere in the middle of while ghosts can go to either side to spawn in as a skeleton knight with the sole purpose of killing anything that isn't their own team. The crew must survive the onslaught of undead knights and make their way to the lich wizard's den to either sneak past the lich to escape through a portal, or outright kill him for both a way home and a random magic-item drop. 

TODO:
- A way to disable the spawners for the crew to end the eternal war once and for all (still need to add the removal part)
- A mount to ride into glorious combat
- Add more to the actual map in terms of detail
- Some sort of environmental/NPC danger if the ghost spawn-count is too low to pose any real player threat.

### CODE REVIEW

**Comment by PKPenguin321** [comment]
crossbow can probably be a child of the harpoon gun

also, >ice rock
i was hoping it would be in the castle itself :(

**Comment by Niknakflak** [comment]
Those are pretty shitty kingdom names

**Comment by RemieRichards** [comment]
I could make a boss mob, do you want me to make a boss mob @MMMiracles ? (Actually making releasing the two factions something decently interesting/difficult)

**Comment by MMMiracles** [comment]
@RemieRichards If you're willing to create a boss for it, I'd like something similar to summoning the wizard that originally cast the curse on the two kingdoms.

**Comment by RemieRichards** [comment]
@MMMiracles was exactly what I was thinking :)

**Comment by Ultimate-Chimera** [comment]
Yes :+1:

**Comment by RemieRichards** [comment]
@MMMiracles PR'd to your branch

**Comment by Bawhoppen** [comment]
thats as good of a fantasy name as glorp

**Comment by MMMiracles** [comment]
you're a glorp

**Comment by Cheridan** [comment]
glorp glorp

File: code/modules/awaymissions/mission_code/castle.dm [hunk 1]
```diff
@@ -0,0 +1,378 @@
...
+//CTF code shamelessly ripped to fit the need for infinite-spawning knights thanks kor also please don't hurt me
+#define WHITE_TEAM "nomad"
+#define TOTH_TEAM "Totharene Kingdom"
+#define YATH_TEAM "Yatherean Kingdom"
+#define DEFAULT_RESPAWN 300 //30 seconds
+
+
+/obj/machinery/knight
+	name = "Portal"
+	desc = "A mystical portal that seems to be the source of never-end knights."
```
Comment (Bawhoppen) [inline code/modules/awaymissions/mission_code/castle.dm]: never-end knights

**Comment by MMMiracles** [comment]
This isn't dead, I've been occupied with other shit. I swear it'll be done eventually™ I promise.

**Comment by MMMiracles** [comment]
Yeah with complaints of already-existing missions still being unbalanced and my motivation to keep working on this steadily declining, I think I'll just put this on backlog for a rainy day.

I already have some generic away-mission stuff I made for the level not commited yet, like spike pits and dart traps, so I'll probably end up chopping those out and placing them in a separate PR soonish for anyone else to use.

**Pull request closed without merging**